sqlmesh/core/engine_adapter/spark.py

```diff
@@ -50,37 +51,141 @@ def _use_spark_session(self) -> bool:
         return True
 
     @classmethod
-    def convert_columns_to_types_to_pyspark_schema(
+    @t.overload
+    def convert_between_spark_and_sqlglot(
```

Comment:
why is this mixing both? it's a much bigger method and makes the types more complicated? just make it one method each.

def spark_to_sqlglot_types

def sqlglot_to_spark_types

Reply:
my guess is you were trying to limit imports of pyspark mapping to one method? just put the import into a method and call it in both

Reply:
done